### DISCUSSION TITLE
Should I prefer status.update for updating many variables for performance reason?

### DISCUSSION THREAD

**Discussion by fdinhSG** [Kitware/trame#257] [Q&A] [answered]
I saw about the possibility of using `state.update()` for updating many variables at once. Is this faster than updating every single variables, one by one? In other words, is the second method preferable?

```python
# Method one
state.a=1
state.b=2
state.c=3

# Method two
state.update({'a': 1, 'b': 2, 'c': 3})
```

**Comment by jourdain** [comment, marked as answer]
They are both as performant. It is just up to the user to pick the syntax they rather use for code clarity. 

The reason why they are the "same" is that the flush (network call) happen after your code execution.

**Reply by fdinhSG** [reply]
Got it, thanks!
